### ISSUE TITLE
linux_user: when modifying a users' homedir, tries to create it even if it exists

### ISSUE THREAD

**Issue by jaymzh** [chef/chef#9080]
## Description

When `linux_user` modifies a user to set it's homedir, if that homedir exists, it tries to create it, fails, and fails the run.

It looks like this is from passing -m always

## Chef Version
14.13.11

## Platform Version
Ubuntu 18.04

## Replication Case
setup a user. change it's homedir to something that exists.

## Client Output
```
---- Begin output of ["usermod", "-d", "/var/lib/openvpn/chroot", "-m", "nm-openvpn"]
 ----
STDOUT: 
STDERR: usermod: directory /var/lib/openvpn/chroot exists
---- End output of ["usermod", "-d", "/var/lib/openvpn/chroot", "-m", "nm-openvpn"] -
---
Ran ["usermod", "-d", "/var/lib/openvpn/chroot", "-m", "nm-openvpn"] returned 12
```

**Comment by skippyj** [comment]
Based on what I know about the "usermod" command, that message is purely informational and should not be evaluating the "<x> directory already exists" message as an error. The manpage of usermod specifically addresses what happens if the passed "-d" directory already exists: https://linux.die.net/man/8/usermod 

tl;dr chef is not actually attempting to re-create the directory, it is simply reading the resulting message as an error when the directory already exists. Without too much looking (on my part) it's possible that the usermod command, despite being informational, is actually returning an err exit code.

**Comment by jaymzh** [comment]
Yeah, it's def `usermod`. But the question is can we handle this better. The man page doesn't list the exit codes, but if 12 is unique to this, then we could set the allowed exit codes to `[0,12]`.

**Comment by skippyj** [comment]
Yeah I agree. I haven't actively contributed to this project before, but I'll take a crack at it.

**Comment by skippyj** [comment]
For reference, the entry reference in usermod that corresponds to code 12: https://github.com/shadow-maint/shadow/blob/b49712ed328ded0cd8161542ca13cf5d4cf55e5f/src/usermod.c#L92

"/* unable to complete home dir move */"

**Comment by skippyj** [comment]
Created branch on my fork
https://github.com/skippyj/chef/commit/768d0f18a091622902e7eefcb6b2f68bc17d3145

**Comment by skippyj** [comment]
Now that the fix is merged, can this be closed or are there further verification steps?

**Comment by jaymzh** [comment]
Fixed by attached PRs
